### PR TITLE
Add general design goals doc

### PR DESCRIPTION
**Pull request by njhale**
**Description of the change:**

- Add a doc to express some general design goals for new features
- Rename `doc` directory to `docs`

### CODE REVIEW

File: docs/design/design-goals.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# General Design Goals
+
+Whether designing new features or redesigning old ones, OLM should maintain these general design goals:
+
+## Build (reasonably) small, sharp features
+
+All designs should be kept as simple and tightly scoped as is reasonably possible. Where possible, attempt to follow the [unix philosophy](https://en.wikipedia.org/wiki/Unix_philosophy) for writing small, sharp features that compose well. Additionally, when writing a new proposal, always remember to [keep it simple ...student!](https://en.wikipedia.org/wiki/KISS_principle)
```
Comment (kevinrizza) [inline docs/design/design-goals.md]: Just thinking out loud: Should we have a more formal process wrt design proposals here? What about https://github.com/operator-framework/enhancements ?

**Comment by ecordell** [comment]
> Rename doc directory to docs

this will require some link updates in places we don't own. @adellape do you think you know where these might be?

**Comment by adellape** [comment]
> this will require some link updates in places we don't own. @adellape do you think you know where these might be?

@ecordell There's a couple in the `openshift-docs` repo that I can update. Other than that, looked in the `openshift/console` repo and only found https://github.com/openshift/console/blob/master/frontend/packages/operator-lifecycle-manager/src/components/clusterserviceversion.tsx#L487. Were there other places that came to mind?

File: docs/design/design-goals.md (hunk 1 above)
Comment (njhale) [reply docs/design/design-goals.md]: I agree -- but I want to keep this doc more abstract. Could we add it in a separate `dev-process` doc?

**Comment by njhale** [comment]
@ecordell I removed the doc rename for now. PTAL.

**Comment by njhale** [comment]
/retest

retesting docs 😅

**Pull request merged**